### DISCUSSION TITLE
Unraid Authelia 4.38 Docker Started No Issues But Apps Are Blocked Using SWAG

### DISCUSSION THREAD

**Discussion by mlause** [authelia/authelia#7036] [Troubleshooting] [answered]
Hi All,

I recently updated to 4.38. The configuration.yml is fine as the Authelia docker starts. I cannot access any apps. When I launch the webui, I get this message: There was an issue retrieving the current user state. None of my reverse proxied apps that have Authelia enabled go through. I'm getting a 500 error. When I disable Authelia on those apps, they work. 

SWAG is on the latest version with the latest authelia-server.conf. The only thing I have noticed that was not an issue before is the docker name is capitalized Authelia. SWAG makes references to lower case authelia. 

What should I check?

M

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Hello M,

Given the information you've provided, there are a few areas we can check to troubleshoot the issue you're experiencing after updating to Authelia version 4.38. The "500 error" and the message "There was an issue retrieving the current user state" suggest there might be a problem with the communication between SWAG (Secure Web Application Gateway) and Authelia, or an issue within Authelia's configuration itself.

1. **Docker Container Name Case Sensitivity**: You mentioned that the Docker container name for Authelia is capitalized ("Authelia") whereas SWAG references it in lowercase ("authelia"). Docker container names are case-sensitive. This discrepancy could potentially lead to issues if SWAG is unable to correctly communicate with the Authelia container due to the case mismatch. Ensure that the container name used in SWAG's configuration matches exactly with the Authelia container name. If necessary, adjust the container name in your `docker-compose.yml` or the relevant SWAG configuration files to ensure consistency.

2. **Check Authelia Logs**: Inspect the logs of the Authelia container for any errors or warnings that could indicate what's going wrong. You can view the logs by running `docker logs Authelia` (replace "Authelia" with the exact name of your container if different). Look for any messages that could point to configuration issues or connectivity problems with the database or Redis (if you're using it).

3. **Review Authelia Configuration**: Ensure that your `configuration.yml` for Authelia is correctly set up, especially the sections related to session management, storage, and the reverse proxy. Since you've updated to a newer version, verify that there are no new or deprecated configuration options that need to be addressed. The official Authelia documentation can be helpful here.

4. **Network Connectivity**: Verify that the network setup in Docker allows SWAG and Authelia to communicate effectively. If you're using Docker networks, ensure that both containers are on the same network and that there are no firewall rules blocking the communication between them.

5. **SWAG Configuration**: Double-check the SWAG configuration, especially the `authelia-server.conf` file, to ensure it's correctly set up to communicate with Authelia. Make sure the URLs and ports specified match those exposed by the Authelia container.

6. **Version Compatibility**: Although it seems you're using the latest versions of both SWAG and Authelia, it's worth checking the release notes or known issues for both applications to see if there are any known compatibility issues or bugs with the versions you're using.

If after checking these areas the issue persists, consider providing more detailed logs or configuration snippets (without sensitive information) for further assistance. Remember to restart the containers after making any configuration changes to ensure they take effect.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by mlause** [comment]
The container case was not an issue on 4.37. I changed the container case to authelia. Now when I enable Authelia on the proxy.conf for an application, it is not working. It's as if Authelia is off (this was the case for home assistant). When I tried enabling Authelia on another app, I get the same 500 error.  The Authelia logs don't have any errors. The configuration.yml seems fine. SWAG and Authelia are on the same network. SWAG is configured correctly. I've searched online and can't seem to find a resolution.

**Comment by james-d-elliott** [comment]
Probably check the logs for both authelia and swag and post them, knowing the exact version is important for us, since you updated the swag configuration it'd be good to know if you updated the Authelia session configuration too and what it looks like.

**Reply by mlause** [reply]
There is nothing in either of the SWAG or Authelia logs. I won't post my SWAG log because it IDs my domains. Here is Authelia.

time="2024-03-27T13:59:04-05:00" level=info msg="Authelia v4.38.6 is starting"
time="2024-03-27T13:59:04-05:00" level=info msg="Log severity set to info"

I'm going to post my authelia config as well.
theme: dark
server:
  address: tcp://:9091/authelia
  buffers:
    read: 4096
    write: 4096
  endpoints:
    authz:
      auth-requests:
        implementation: AuthRequest
log:
  level: info
  file_path: /config/logs/authelia.log
identity_validation:
  reset_password:
    jwt_secret: secret
authentication_backend:
  password_reset:
    disable: false
  file:
    path: /config/users_database.yml
    password:
      algorithm: argon2
      argon2:
        variant: argon2id
        iterations: 1
        key_length: 32
        salt_length: 16
        memory: 512
        parallelism: 8
access_control:
  default_policy: deny
  networks:
    - name: internal
      networks: 
        - 10.10.0.0/16
  rules:
    - domain:
      - 'mydomain.com'
      resources:
      - 'resource'
      policy: two_factor
    - domain:
      - 'mydomain.com'
      policy: bypass
    - domain: 
      - 'mydomain.com'
      resources: 
      - 'resource'
      policy: bypass
    - domain:
      - 'mydomain.com'
      resources:
      - 'resource'
      policy: two_factor
    - domain:
      - 'mydomain.com'
      policy: bypass
session:
  secret: secret
  name: authelia_session
  same_site: lax
  expiration: 1h
  inactivity: 5m
  remember_me: 5m
  cookies:
    - domain: mydomain.com
      authelia_url: https://auth.mydomain.com
      default_redirection_url: https://mydomain.com
  redis:
    host: ip
    port: 6379
    password: pw
    database_index: 1
    maximum_active_connections: 10
    minimum_idle_connections: 0
regulation:
  max_retries: 5
  find_time: 2m
  ban_time: 10m
storage:
  encryption_key: key
  mysql:
    address: tcp://ip:3306
    database: authelia
    username: authelia
    password: pw
notifier:
  disable_startup_check: false
  smtp: 
    username: me@mydomain.com
    password: pw
    address: submission://smtp.emailprovider.com:587
    sender: me@mydomain.com
    subject: "[Authelia] {title}"
    startup_check_address: test@authelia.com
    disable_require_tls: false
    tls:
      skip_verify: false
      minimum_version: TLS1.2

**Reply by james-d-elliott** [reply]
The authelia logs indicate no requests are making it to Authelia, what happens when you visit auth.mydomain.com?

**Reply by mlause** [reply]
I get this message, Welcome to your [SWAG](https://github.com/linuxserver/docker-swag) instance. If I pull the local IP, I get this message, There was an issue retrieving the current user state. This never happened with 4.37. So I am not sure what change in 4.38 to cause this issue.

**Reply by james-d-elliott** [reply]
No one else has had this specific issue excluding when SWAG was misconfigured, and has occurred prior to 4.38, it's unlikely to be a 4.38 issue. This is also evident by the fact visiting the URL directly doesn't correctly reverse proxy you to the container (and your SWAG configuration was changed).

If you're going to make assumptions we probably wont get very far. 

Try deleting the two authelia-*.conf files and the proxy.conf file (or rename them), correct the container name issue, restart SWAG, and show the server configuration for auth.mydomain.com.

**Reply by mlause** [reply]
Within SWAG, I didn't have an authelia.subdomain.conf file. I never had it before when it worked. So I added one, and I can now see Authelia by going to https://auth.mydomain.com. However, when enabled for an application it still doesn't work.

**Reply by mlause** [reply]
It seems we sent a message at the same time. Here is what I have done.

1. Deleted authelia-location.conf and authelia-server.conf and restarted SWAG. The files were replaced with new ones.
2. Added authelia.subdomain.conf. I can now see https://auth.mydomain.com. 
4. Deleted the Authelia docker and rebuilt it with lowercase authelia.
4. When an application proxy.conf has Authelia enabled by uncommenting this line:
# enable for Authelia (requires authelia-server.conf in the server block)
        include /config/nginx/authelia-location.conf;
I get the 500 error. If I comment otu authelia, it works. So there is some commincation gap happening between SWAG and Authelia.

**Reply by james-d-elliott** [reply]
The 500 error reason should be reflected somewhere in the swag error logs, can you double check it? Something really odd is occurring.

**Reply by mlause** [reply]
This is what I have in the most recent line in the SWAG error log.

2024/03/27 18:13:11 [error] 692#692: *378 auth request unexpected status: 404 while sending to client, client: router_ip_address, server: myapp.*, request: "GET /api/shares HTTP/2.0", host: "myapp.mydomain.com"

**Reply by james-d-elliott** [reply]
Can you enable trace logs for Authelia? Looks like either it's requesting from another service or the wrong path based on your config. I think the cause is this (you've configured `/api/authz/auth-requests` and SWAG uses `/api/authz/auth-request` by default):

```yaml
server:
  endpoints:
    authz:
      auth-requests:
        implementation: AuthRequest
```

**Reply by mlause** [reply]
I can change the log level from info to trace, which I did, but I can't read the log file. 

I've change it to 
```
server:
  endpoints:
    authz:
      auth-request:
        implementation: AuthRequest
```
and I still get a 500 error.

**Reply by james-d-elliott** [reply]
What do you mean you can't read it?

**Reply by mlause** [reply]
Console app on my Mac can’t read it. If I copy it and try to open it with Textmate, it’s blank. I’m not sure if this has anything to do with Unraid or my setup.

**Reply by james-d-elliott** [reply]
Try setting the following then it'll ensure the docker logs show it as well and we can eliminate FS issues pretty reliably:

```yaml
log:
  level: trace
  file_path: /config/logs/authelia.log
  keep_stdout: true
```

**Reply by mlause** [reply]
That works at showing the server log. When I to to https://auth.mydomain.com I see a bunch of activity. Going to one of my apps with authelia enabled, I get nothing. Otherwise I get something like this like every so many seconds.

time="2024-03-27T23:38:28-05:00" level=trace msg="Replied (status=200)" method=GET path=/api/health remote_ip=127.0.0.1

**Reply by james-d-elliott** [reply]
Think this one will be easier to fix via chat. Do you have Matrix or Discord and feel comfortable with joining? There is something preventing the auth_request from making it to authelia itself and the fact it's not logging errors indicates nginx never makes the request to the container port which is very peculiar.

**Reply by mlause** [reply]
I have a discord account. I don't have an issue joining. How would I do that?

**Reply by james-d-elliott** [reply]
https://discord.authelia.com

**Comment by james-d-elliott** [comment, marked as answer]
Was the `include /config/nginx/authelia-server.conf;` being commented in the affected applications subdomain.conf, easy to miss. But the symptoms above potentially match that solution.

**Comment by faninx** [comment]
Rolling back the authelia-location.conf and authelia-server.conf configuration files to the previous version (not compatible with Authelia 4.38) can temporarily solve this problem.

**Reply by james-d-elliott** [reply]
The solution is to use the new versions and ensure the  includes are both uncommitted.
